createcourse/tasks.py

```diff
@@ -49,19 +53,19 @@ def fetch_data():
             #Course basic info data is prepared now join with learning module
             with open(
                     os.path.join(
-                        'fixtures/',
+                        'fixtures',
                         workshop_json_file_mapper[str(course_duration)])) as f:
                 json_data = json.load(f)
             course_info.update(
                 json_data
             )  #learning module data is appended in course_info dict
             post_response = requests.post(yaksh_post_url, json=course_info)
-            post_response.raise_for_status()
             if post_response.status_code == 201:
                 #update the workshop cached with a new entry.
                 cached_workshop.status = 1
                 cached_workshop.save()
             else:
+                cached_workshop.status = 2
```

Comment:
This will not save the data in the database. You did not do `cached_workshop.save()`.
Also, save the status as 2 in the except conditions below too.

Reply:
You did not save the status in the except condition for `HTTPError` and `Exception`.